packages/teleport-shared/src/utils/ast-js-utils.ts

```diff
@@ -111,3 +112,85 @@ export const getTSAnnotationForType = (type: any, t = types) => {
       return t.tsUnknownKeyword()
   }
 }
+
+export const createMethodsObject = (
+  methods: Record<string, UIDLEventHandlerStatement[]>,
+  propDefinitions: Record<string, UIDLPropDefinition>,
+  flavor: 'angular' | 'vue',
```

Comment:
this function is not truly shareable, since it has the flavor dependency. I would handle everything related to the specifics of vue and angular in their base plugins and leave here in shared these functions:

createEventHandlerBody
createStateChangeStatement
createPropCallStatement

Reply:
Yeah, felt the same but wanted to test the solution first if it really works. Now I will clean this up 👍 :D